src/unlimited.cpp

```diff
@@ -1824,31 +1826,115 @@ UniValue getstat(const UniValue &params, bool fHelp)
+    {"mining", "setminingmaxblock", &setminingmaxblock, true},
+    {"mining", "getminercomment", &getminercomment, true},
+    {"mining", "setminercomment", &setminercomment, true},
+    {"mining", "getblockversion", &getblockversion, true},
     {"mining", "setblockversion", &setblockversion, true},
+    {"mining", "validateblocktemplate",  &validateblocktemplate, true},
 
     /* Utility functions */
-    {"util", "getstatlist", &getstatlist, true}, {"util", "getstat", &getstat, true}, {"util", "get", &gettweak, true},
+    {"util", "getstatlist", &getstatlist, true},
+    {"util", "getstat", &getstat, true},
+    {"util", "get", &gettweak, true},
     {"util", "set", &settweak, true},
 
     /* Coin generation */
-    {"generating", "getgenerate", &getgenerate, true}, {"generating", "setgenerate", &setgenerate, true},
+    {"generating", "getgenerate", &getgenerate, true},
+    {"generating", "setgenerate", &setgenerate, true},
+
 };
+/* clang-format on */
 
 void RegisterUnlimitedRPCCommands(CRPCTable &tableRPC)
 {
     for (unsigned int vcidx = 0; vcidx < ARRAYLEN(commands); vcidx++)
         tableRPC.appendCommand(commands[vcidx].name, &commands[vcidx]);
 }
+
+
+UniValue validateblocktemplate(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 1)
+        throw runtime_error(
+            "validateblocktemplate \"hexdata\"\n"
+            "\nReturns whether this block template will be accepted if a hash solution is found.\n"
+            "The 'jsonparametersobject' parameter is currently ignored.\n"
+            "See https://en.bitcoin.it/wiki/BIP_0022 for full specification.\n"
+
+            "\nArguments\n"
+            "1. \"hexdata\"    (string, required) the hex-encoded block to validate (same format as submitblock)\n"
+            "\nResult:\n"
+            "true (boolean) submitted block template is valid\n"
+            "JSONRPCException if submitted block template is invalid\n"
+            "\nExamples:\n"
+            + HelpExampleCli("validateblocktemplate", "\"mydata\"")
+            + HelpExampleRpc("validateblocktemplate", "\"mydata\"")
+        );
+
+    UniValue ret(UniValue::VARR);
+    CBlock block;
+    if (!DecodeHexBlk(block, params[0].get_str()))
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Block decode failed");
+
+    if (block.nBlockSize == 0) block.nBlockSize = ::GetSerializeSize(block, SER_NETWORK, PROTOCOL_VERSION);
+
+    CBlockIndex *pindexPrev=NULL;
+    {
+        LOCK(cs_main);
+
+        BlockMap::iterator i = mapBlockIndex.find(block.hashPrevBlock);
+        if (i == mapBlockIndex.end())
+        {
+            throw runtime_error("invalid block: unknown parent");
+        }
+
+        pindexPrev = i->second;
+
+        if (pindexPrev != chainActive.Tip())
+        {
+            throw runtime_error("invalid block: does not build on chain tip");
+        }
```

Comment:
this is just an RPC call that checks if an unsolved block passed to this RPC is valid.  It would be great if the code could determine whether a block that's on another fork is valid.  Unfortunately, the code is incapable of doing this, because there is just one UTXO set.  So the RPC call returns an error.  
But this RPC call has no effects on subsequent submission of solved blocks...